cds/modules/records/tasks.py

```diff
@@ -223,3 +228,108 @@ def file_integrity_report():
+    report = []
+    cache = current_cache.get('subformats_integrity:task_details')
+    task_details = cache if cache else {}
+
+    if 'last_run' not in task_details:
+        task_details['last_run'] = datetime.utcnow()
+
+    record_uuids = (
+        RecordMetadata.query
+        .join(RecordsBuckets)
+        .join(Bucket)
+        .join(ObjectVersion)
+        .filter(ObjectVersion.created > task_details['last_run'])
+        .group_by(RecordMetadata.id).with_entities(RecordMetadata.id)
+        .all())
+
+    records = (CDSRecord.get_record(record.id) for record in record_uuids)
+    for record in records:
+        master = CDSVideosFilesIterator.get_master_video_file(record)
+
+        if not master:
+            report.append({
+                'message': 'No master file',
+                'recid': record['recid'],
+                'report_number': record['report_number'][0]})
+            continue
+
+        master_obj = as_object_version(master['version_id'])
+        try:
+            probe = ff_probe_all(master_obj.file.uri.replace(
+                'root://eosmedia.cern.ch/', ''))
+
+            if not probe.get('streams'):
+                report.append({
+                    'message': 'No video streams',
+                    'recid': record['recid'],
+                    'report_number': record['report_number'][0],
+                    'other': str(master_obj)})
+
+        except Exception as e:
+            report.append({
+                'message': 'No video streams',
+                'recid': record['recid'],
+                'report_number': record['report_number'][0],
+                'other': str(master_obj),
+                'error': repr(e)})
+
+        subformats = CDSVideosFilesIterator.get_video_subformats(master)
+        if not subformats:
+            report.append({
+                'message': 'No subformats found',
+                'recid': record['recid']})
+        for subformat in subformats:
+            subformat_obj = as_object_version(subformat['version_id'])
+            try:
+                probe = ff_probe_all(subformat_obj.file.uri.replace(
+                    'root://eosmedia.cern.ch/', ''))
+
+                if not probe.get('streams'):
+                    report.append({
+                        'message': 'No video streams',
+                        'recid': record['recid'],
+                        'report_number': record['report_number'][0],
+                        'other': str(subformat_obj)})
+
+            except Exception as e:
+                report.append({
+                    'message': 'Error while running ff_probe_all',
+                    'recid': record['recid'],
+                    'report_number': record['report_number'][0],
+                    'other': str(master_obj),
+                    'error': repr(e)})
+
+    current_cache.set(
+        'subformats_integrity:task_details', task_details, timeout=-1)
+
+    if report:
```

Comment:
Looks very similar to previous task this one, maybe you can avoid the duplication, so next tasks we will use it again.